OptimoveSDK/optimove-sdk/src/main/java/com/optimove/android/preferencecenter/OptimovePreferenceCenter.java

```diff
@@ -0,0 +1,300 @@
+package com.optimove.android.preferencecenter;
+
+import android.os.Handler;
+import android.os.Looper;
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.optimove.android.OptimoveConfig;
+import com.optimove.android.main.tools.networking.HttpClient;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import okhttp3.Response;
+
+import java.io.IOException;
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+
+public class OptimovePreferenceCenter {
+    private static final String TAG = OptimovePreferenceCenter.class.getName();
+    private static OptimovePreferenceCenter shared;
+    private static OptimoveConfig config;
+    private static @Nullable String customerId;
```

Comment:
what if customer id changes? what i was proposing is not to store it here at all, not pass it during initialisation, but read from user info when request is done

Reply:
yes noticed this wasn't right an working on changes now 👍

Reply:
1. passing UserInfo reference like this i think is very not safe. If someone does `userInfo = new UserInfo()` somewhere -- it breaks PC silently. it's also not explicit / hard to see how it is updated / how many references where. We could just keep it in Optimove and do `Optimove.getInstance().getUserInfo();` without passing any arg
2. the problem is that each of 3 features can exist independently theoretically. UserInfo it turns out is currently linked to Optimove. we should set it even if optimove is not enabled -- but let's do a little separate pr after this one

Reply:
@scotthowie-optimove okay, as on slack, looks optimove becomes "core" and we assume it's always there. Then we can just assume userInfo and tenantInfo are always set

Reply:
Is it worth moving the preference center initialization in `Optimove.java` inside the if block checking if `isOptimoveConfigured`?

Reply:
mb similar (same?), but a little neater is smth like

```
 if (config.isOptimoveConfigured() && config.isPreferenceCenterConfigured()) {
            OptimovePreferenceCenter.initialize(config, shared.userInfo, shared.tenantInfo.getTenantId());
        }
```

Reply:
kinda maybe it highlights that in current implementation they are separate features (so 3 blocks) , but 1 requires another. idk, mb ~same :D